tests/baselines/reference/enumBasics.js

```diff
@@ -145,7 +145,7 @@ var E8;
 var E9;
 (function (E9) {
     E9[E9["A"] = 0] = "A";
-    E9[E9["B"] = 0] = "B";
+    E9["B"] = E9.A;

```

Comment:
It's a simpler model to work with (that the original declaration doesn't change).
```
// lib.ts
enum  foo {
a = 1
b = 2
}

// thirdparty.ts
enum foo {
c = 1
}
```
Why is 'thirdparty.ts' allowed to modify the reverse mapping?

About string enums, I've implemented it here:
https://github.com/Microsoft/TypeScript/compare/master...jbondc:setTypes-v2

There's some wonkyness still but overall think it will scale well with literals/set type guards.